pkg/securelaunch/helpers.go

```
@@ -46,49 +47,72 @@ var StorageBlkDevices block.BlockDevices
 // kernel cmdline is checked in sluinit.
 var Debug = func(string, ...interface{}) {}
 
-// WriteToFile writes a byte slice to a file on an already mounted disk and
-// returns the file path written to.
-func WriteToFile(data []byte, dst, defFileName string) (string, error) {
-	// make sure dst is an absolute file path
-	if !filepath.IsAbs(dst) {
-		return "", fmt.Errorf("dst =%s Not an absolute path ", dst)
+// ReadFile reads a file into a byte slice. It mounts the disk if necessary.
+//
+// policyLocation is formatted as `<block device id>:<path>`
+//
+//	e.g., sda1:/boot/securelaunch.policy
+//	e.g., 4qccd342-12zr-4e99-9ze7-1234cb1234c4:/foo.txt
+func ReadFile(fileLocation string) ([]byte, error) {
+	mountedFilePath, err := GetMountedFilePath(fileLocation, mount.MS_RDONLY)
+	if err != nil {
+		return nil, err
 	}
 
-	// target is the full absolute path where []byte will be written to
-	target := dst
-	dstInfo, err := os.Stat(dst)
-	if err == nil && dstInfo.IsDir() {
-		Debug("No file name provided. Adding it now. old target=%s", target)
-		target = filepath.Join(dst, defFileName)
-		Debug("New target=%s", target)
-	}
+	Debug("ReadFile: reading file '%s' (mounted at '%s')", fileLocation, mountedFilePath)
 
-	Debug("WriteToFile: target=%s", target)
-	err = os.WriteFile(target, data, 0o644)
+	fileBytes, err := os.ReadFile(mountedFilePath)
 	if err != nil {
-		return "", fmt.Errorf("failed to write date to file =%s, err=%v", target, err)
+		return nil, fmt.Errorf("could not read file '%s' (mounted at '%s')", fileLocation, mountedFilePath)
```

Why not return the wrapper error? You were using err before?